tests/scripts/thread-cert/mesh_cop.py

```diff
@@ -0,0 +1,832 @@
+    # TODO: Not implemented yet
+
+    def parse(self, data, message_info):
+        raise NotImplementedError("TODO: Not implemented yet")
+
+
+# ChannelMask TLV (53)
+class ChannelMask(object):
+    # TODO: Not implemented yet
+    pass
+
+
+class ChannelMaskFactory(object):
+    # TODO: Not implemented yet
+
+    def parse(self, data, message_info):
+        raise NotImplementedError("TODO: Not implemented yet")
+
+
+# Count TLV (54)
+class Count(object):
+    # TODO: Not implemented yet
+    pass
+
+
+class CountFactory(object):
+    # TODO: Not implemented yet
+
+    def parse(self, data, message_info):
+        raise NotImplementedError("TODO: Not implemented yet")
+
+
+# Period TLV (55)
+class Period(object):
+    # TODO: Not implemented yet
+    pass
+
+
+class PeriodFactory(object):
+    # TODO: Not implemented yet
+
+    def parse(self, data, message_info):
+        raise NotImplementedError("TODO: Not implemented yet")
+
+
+# ScanDuration TLV (56)
+class ScanDuration(object):
+    # TODO: Not implemented yet
+    pass
+
+
+class ScanDurationFactory(object):
+    # TODO: Not implemented yet
+
+    def parse(self, data, message_info):
+        raise NotImplementedError("TODO: Not implemented yet")
+
+
+# EnergyList TLV (57)
+class EnergyList(object):
+    # TODO: Not implemented yet
+    pass
+
+
+class EnergyListFactory(object):
+    # TODO: Not implemented yet
+
+    def parse(self, data, message_info):
+        raise NotImplementedError("TODO: Not implemented yet")
+
+
+
+class MeshCopCommand(object):
+
+    def __init__(self, _type, tlvs):
+        self._type = _type
+        self._tlvs = tlvs
+
+    @property
+    def type(self):
+        return self._type
+
+    @property
+    def tlvs(self):
+        return self._tlvs
+
+    def __repr__(self):
+        tlvs_str = ", ".join(["{}".format(tlv) for tlv in self.tlvs])
+        return "MeshCopCommand(type={}, tlvs=[{}])".format(self.type, tlvs_str)
+
+def create_deault_mesh_cop_msg_type_map():
+    return {
+        'JOIN_FIN.req': MeshCopMessageType.JOIN_FIN_REQ,
+        'JOIN_FIN.rsp': MeshCopMessageType.JOIN_FIN_RSP,
+        'JOIN_ENT.ntf': MeshCopMessageType.JOIN_ENT_NTF,
+        'JOIN_ENT.rsp': MeshCopMessageType.JOIN_ENT_RSP
+    }
+
+class MeshCopCommandFactory:
+
+    def __init__(self, tlvs_factories):
+        self._tlvs_factories = tlvs_factories
+        self._mesh_cop_msg_type_map = create_deault_mesh_cop_msg_type_map()
+
+    def _get_length(self, data):
+        return ord(data.read(1))
+
+    def _get_tlv_factory(self, _type):
+        try:
+            return self._tlvs_factories[_type]
+        except KeyError:
+            raise KeyError("Could not find TLV factory. Unsupported TLV type: {}".format(_type))
+
+    def _parse_tlv(self, data):
+        _type = TlvType(ord(data.read(1)))
+        length = self._get_length(data)
+        value = data.read(length)
+        factory = self._get_tlv_factory(_type)
+        if factory == None:
+            return None
```

Comment:
factory will not be `None`, should catch the exception.

Reply:
Yeah, actually `_get_tlv_factory` will catch the exception.